specification/purview/data-plane/Azure.Analytics.Purview.DataPolicies/preview/2023-08-01-preview/purviewDataPolicy.json

```
@@ -0,0 +1,616 @@
+            "Group",
+            "ServicePrincipal"
+          ],
+          "x-ms-enum": {
+            "name": "objectType",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "FabricItemMember": {
+      "type": "object",
+      "description": "Microsoft fabric item member",
+      "properties": {
+        "sourcePath": {
+          "type": "string",
+          "description": "Source path of the member"
+        },
+        "itemAccess": {
+          "type": "array",
+          "description": "Array of access items for the member",
+          "items": {
+            "type": "string",
+            "description": "access item"
+          }
+        }
+      }
+    },
+    "PolicyDecisionRule": {
+      "type": "object",
+      "description": "Purview RBAC policy decision rule",
+      "required": [
+        "effect"
+      ],
+      "properties": {
+        "effect": {
+          "description": "The effect for rule",
+          "enum": [
+            "Permit",
+            "Deny"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "Decision",
+            "modelAsString": true
+          }
+        },
+        "permission": {
+          "type": "array",
+          "description": "Array of attribute predicates",
+          "items": {
+            "$ref": "#/definitions/AttributePredicate"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "AttributePredicate": {
+      "type": "object",
+      "description": "Attribute predicate for a policy permission",
+      "properties": {
+        "attributeName": {
+          "type": "string",
+          "description": "AttributeName of a policy permission"
+        },
+        "attributeValueIncludedIn": {
+          "type": "array",
+          "description": "values for a policy permission",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "api-version": {
+      "in": "query",
+      "name": "api-version",
+      "description": "The api version to use.",
+      "required": true,
+      "type": "string",
+      "x-ms-client-name": "ApiVersion",
+      "x-ms-parameter-location": "client"
+    },
+    "endpoint": {
+      "in": "path",
+      "name": "endpoint",
+      "description": "The endpoint of your Purview account. Example: https://{accountName}.purview.azure.com.",
+      "required": true,
+      "type": "string",
+      "format": "url",
+      "x-ms-parameter-location": "client",
+      "x-ms-skip-url-encoding": true
+    },
+    "policy-name": {
+      "in": "path",
+      "name": "policy-name",
+      "description": "Name of the policy to be deleted.",
+      "required": true,
+      "type": "string"
+    },
+    "if-match": {
+      "in": "header",
+      "name": "If-Match",
+      "description": "Http Etag value of a policy to be updated. Required only during the replace operation",
+      "type": "string"
```

I think we may want `format: etag` for this header ... for languages like .NET that have a type for ETags.
```suggestion
      "type": "string",
      "format": "etag"
```

---

If we change to format: etag. It is causing some validation errors. Keeping the type as string.

---

What are the validation errors?

---

https://github.com/Azure/azure-rest-api-specs/runs/15950819967

---

Its 

```
type: string,
format: etag
```

not
```
type: etag
```

---

It now gives below error. "etag is not a valid format"
https://github.com/Azure/azure-rest-api-specs/pull/25059/checks?check_run_id=16064919893

---

Sorry, we were misinformed on this one. `etag` is not (currently) a format recognized by autorest or the openapi-validator. So drop the `format: etag` for now.

---

thanks.